vcd/resource_vcd_nsxt_edgegateway_auto_allocation_test.go

```diff
@@ -0,0 +1,430 @@
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway.nsxt-edge", "dedicate_external_network", "false"),
+					resource.TestCheckResourceAttrSet("vcd_nsxt_edgegateway.nsxt-edge", "primary_ip"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway.nsxt-edge", "used_ip_count", "1"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway.nsxt-edge", "unused_ip_count", "24"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway.nsxt-edge", "total_allocated_ip_count", "25"),
+					resource.TestCheckTypeSetElemNestedAttrs("vcd_nsxt_edgegateway.nsxt-edge", "auto_allocated_subnet.*", map[string]string{
+						"gateway":            "14.14.14.1",
+						"prefix_length":      "24",
+						"allocated_ip_count": "12",
+					}),
+					resource.TestCheckTypeSetElemNestedAttrs("vcd_nsxt_edgegateway.nsxt-edge", "auto_allocated_subnet.*", map[string]string{
+						"gateway":            "93.0.0.1",
+						"prefix_length":      "24",
+						"allocated_ip_count": "13",
+					}),
+				),
+			},
+			{ // update with decrease
+				Config: configText3,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway.nsxt-edge", "name", params["NsxtEdgeGatewayVcd"].(string)),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway.nsxt-edge", "dedicate_external_network", "false"),
+					resource.TestCheckResourceAttrSet("vcd_nsxt_edgegateway.nsxt-edge", "primary_ip"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway.nsxt-edge", "used_ip_count", "1"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway.nsxt-edge", "unused_ip_count", "8"),
+					resource.TestCheckResourceAttr("vcd_nsxt_edgegateway.nsxt-edge", "total_allocated_ip_count", "9"),
+					resource.TestCheckTypeSetElemNestedAttrs("vcd_nsxt_edgegateway.nsxt-edge", "auto_allocated_subnet.*", map[string]string{
+						"gateway":            "14.14.14.1",
+						"prefix_length":      "24",
+						"allocated_ip_count": "5",
+					}),
+					resource.TestCheckTypeSetElemNestedAttrs("vcd_nsxt_edgegateway.nsxt-edge", "auto_allocated_subnet.*", map[string]string{
+						"gateway":            "93.0.0.1",
+						"prefix_length":      "24",
+						"allocated_ip_count": "4",
+					}),
+				),
+			},
+		},
+	})
+	postTestChecks(t)
+}
+
+const testAccNsxtEdgeGatewayAutoAllocatedSubnet = testAccNsxtEdgeGatewayAutoAllocationPrerequisites + `
+resource "vcd_nsxt_edgegateway" "nsxt-edge" {
+  org  = "{{.Org}}"
+  vdc  = "{{.NsxtVdc}}"
+  name = "{{.NsxtEdgeGatewayVcd}}"
+
+  external_network_id = vcd_external_network_v2.ext-net-nsxt.id
+
+  auto_allocated_subnet {
+    gateway       = tolist(vcd_external_network_v2.ext-net-nsxt.ip_scope)[0].gateway
+    prefix_length = tolist(vcd_external_network_v2.ext-net-nsxt.ip_scope)[0].prefix_length
+
+	primary_ip         = tolist(tolist(vcd_external_network_v2.ext-net-nsxt.ip_scope)[0].static_ip_pool)[0].end_address
+	allocated_ip_count = "{{.Subnet1Count}}"
```

Comment:
Nit. GitHub shows these two lines indented too much. But maybe it's just GitHub...

Reply:
It was "tabs vs spaces" - I have fixed it.